templates/core/core-dpl.yaml

```diff
@@ -96,6 +96,20 @@ spec:
           - name: INTERNAL_TLS_TRUST_CA_PATH
             value: /etc/harbor/ssl/core/ca.crt
          {{- end }}
+         {{- if .Values.database.external.existingSecret }}
+          - name: POSTGRESQL_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.database.external.existingSecret }}
+                key: password
```

Comment:
Since the code comment in values.yaml has been changed to 
```
# if using existing secret, the key must be PASSWORD
```
please use upper case PASSWORD. 
Another preferred way, I would suggest change the `PASSWORD` to `POSTGRESQL_PASSWORD` in both the code comment (in values.yaml) and here:
```
          - name: POSTGRESQL_PASSWORD
            valueFrom:
              secretKeyRef:
                name: {{ .Values.database.external.existingSecret }}
                key: POSTGRESQL_PASSWORD
```

Reply:
For this, I used "password" as this is the key given by the zalando postgresql operator when generating the secret needed to access the database. this allows automatic postgres usage with the auto-generated secret.